### ISSUE TITLE
Nerf mangonel towers and fire mechanics.

### ISSUE THREAD

**Issue by Rakso69** [Sh0wdown/UnofficialCrusaderPatch#64]
As title above. As everyone knows the mangonel towers only defence is pretty much overpowered online, as well as fire spreading is. Type below your own suggestions.
- Myself I would give a limit of 3 mangonel towers and 3 tower balista for every player(equally as the AI has).
- About fire: I'd make the "explosions" have longer time to trigger and spread fire futher, instead of the super fast mechanic We know. I wouldn't change the fire from defensive purposes tho.

**Comment by Lolasik011** [comment]
You are a cool ideas generator ;D

**Comment by Lolasik011** [comment]
I think this can be done optionally in the launcher.

**Comment by ByBurton** [comment]
@Rakso69 I disagree with almost all your ideas here, except that fire needs a nerf.
Tower ballistae and mangonels are okay. 
The AI needs to be able to build up to 5 (fire-) ballistae, tower mangonels and trebuchets, not human players less.

**Comment by Rakso69** [comment]
@ByBurton No. They're not okay. Balistas are, but mangonels in any way. You can't rate their strenght based on AI. Ofc AI is okay, but the mechanic is not, especially when abused. Imagine 40 towers with only mangonels mounted over spammed by someone in online game. The mangonels one-shoot kill everything except for a battering ram and siege tower. Balance is really broken compared to attack when it comes to anything in Stronghold 1 & Crusader. Both in gold&resource cost and presumable effectiveness. It's not good.

Think about the big picture, about the whole game. Don't consider only AI in your ideas and your own satisfaction. Don't be selfish. 

At any case, You'll be able to disable all the disliked features on installation, so it's ok.

**Comment by Monsterfisch** [comment]
Fire rate of mangonel could be reduced and its range could be reduced as well to be shorter than a catapult. That way besieging catapults can sit out of mangonel range and take down those mangonel towers which would only be effective against engaging melee units.
(note range would roughly include the random spread which is one one the biggest problems as it increases mangonel reach additionally past the "actual" selectable reach.

**Comment by brotolisk** [comment]
Can magonels be made to do a lot less damage to buildings and walls?

Especially with the small map sizes of crusader AI frequently build castles within magonel distance of each other. Usually whoever gets their tower up first wins. Whilst the other is crippled

**Comment by Monsterfisch** [comment]
If their range would be reduced I don't think that damage to buildings would be a big issue anymore. particularly if the spread was reduced slightly so the randomness of stone flying super far out of the predicted area wouldn't be happening so often anymore as well.

**Comment by Sh0wdown** [comment]
The easiest thing I could do is too reduce the damage, so it does not one-hit everything. What would you think about this?

I could also try to implement a limit per player.

**Comment by PitchNeeded** [comment]
Don't reduce the damage by too much though.....the two or three mangonels some AIs use should still be effective for them I think!

**Comment by PitchNeeded** [comment]
> If their range would be reduced I don't think that damage to buildings would be a big issue anymore. particularly if the spread was reduced slightly so the randomness of stone flying super far out of the predicted area wouldn't be happening so often anymore as well.

I don't think the range should be reduced for mangonels, I like the fact that mangonels can potentially crush catapults firing at range, if the range was shorter, the defence of AI lords such as Wolf, Frederick, Emir and Lionheart who like using mangonels at the front of their castles would be much weaker and they would really struggle against any catapult fire.

**Comment by Monsterfisch** [comment]
not a fan of an amount limit at all, same with damage. as with defending with mangonels it just doesn't make sense to use them for defense against the siege that's what the ballista is for clearly, the mangonel is more fun but if you want to crush enemy siege with a stone just use trebuchets they have larger range anyways and are one shot kill.

arguably you could make it so that each shot consumes stone from the stockpile but I can see that being an issue for people too. 

as for trebs being useless in comparison, I can't agree as trebs can be placed outside of ballista range what catas cannot do. while on that topic if ballista would always prioritize siege over any other units by default that would make trebs a bunch stronger as they are able to fire from a safe distance.  

I just always saw mangonels as an aoe weapon which in my head shouldn't have a long range as combined with instant death (which is something I would like to keep) is way too powerful.

granted there are many things that could be tried to balance it but I personally think reduced range and slightly reduced spread would be the right direction. as that way, it would be better used to defend against large approaching armies but would need support fire from ballistas to be "safe" against other siegeunits. as it stands right now it just feels like magonels are a 1 box solution as it can defend even against trebs due to the random spread potentially even hitting trebs outside of ballista range. 
Trebs should be defended against using own trebs or shock troops at least in my head.  with the range reduction on the mangonel it would behave like mangonels currently behave towards trebs where they would barely hit the catas with the random spread so it wouldn't be completely pointless to try and defend against them just not as effective as right now.

**Comment by PitchNeeded** [comment]
Reduced range mangonels would be good for multiplayer I can see, but it would be bad for single player as it would make the AI have a weaker defence. Wolf and Frederick would be easy to defeat with catapults :(

I think a reduced fire rate would be the best solution I think, that way they can still deal out instant death as you want them too :D, AIs can still destroy enemy catapults with them as they can now, but it would reduce the overall effectiveness of mangonels a bit so they are not so overpowered.

**Comment by Monsterfisch** [comment]
mhm I see what you mean but apart from Saladin and Richard the other ai lords that use mangonels don't really use them for defense against siege consistently considering they are not precise nor instakill when hitting siege so I personally don't think it would be so heavily affected, though that can vary on playstyle of the player as well I guess.

What one could do to balance the reduction of range would be to increase the spread? so it would somewhat reach the max range you could select with the marker before but only with the random spread. 

Could even potentially entertain the idea of giving some ai lords certain advantages in tech like having increased range for some units like mangonels to make them more competitive. if that is possible with the methods available.

All in all its a tough nut to crack all I know is I don`t like using mangonels since they feel like cheating when playing against ai and I doubt that just making them fire slower or reduce the amounts of them being placeable will change much about that as their range is just if not further than ballista considering the random spread and they even now are a longterm damage  weapon when attacking closebycastles with it which in return will just stretch out the games even further. taking away range makes sense since its alot more projectiles than catas and it should be a weaker machine since its smaller than catas since its mounted on a tower that can only support a certain amount of weight unlike the ground https://i.imgur.com/oMRWpAc.png

**Comment by PitchNeeded** [comment]
Poor Frederick would suffer particularly if there was reduced mangonel range as he doesn't have ballistas :( . Neither does Emir but at least it is easy to give Emir ballistas in the AIV!

> Could even potentially entertain the idea of giving some ai lords certain advantages in tech like having increased range for some units like mangonels to make them more competitive. if that is possible with the methods available.

That would be a good idea actually, just reduce the range of the human player's mangonels. And that would be balanced because of course the AI cannot make as many mangonels as the human player so the AI wouldn't be 'cheating' as such.

**Comment by Panbutt** [comment]
I don't have an opinion on mangolels but I definitely believe that fire -needs- to be nerfed. It spreads way too quickly. An entire castle can catch on fire in less then a second and destroy an economy before well-workers can even react. It needs to spread slower and reasonably. Castle walls also need to be able to block fire. It makes no sense that fire outside a castle can sometimes just phase through the wall and catch the interior a blaze.

**Comment by Monsterfisch** [comment]
fire sure gets out of hand quickly I would suggest making it spread slower and have it be that fire on a building would take more than one bucket to be put out but it would also stop spreading as soon as one bucket has been applied. that being said it still pretty satisfying to see castles burn down so it shouldn't be changed too much.

**Comment by Heroesflorian** [comment]
🗿 **Regarding mangonels** 🗿 
One more tweak not having been mentioned yet would be to modify the amount of stones per volley, if that is possible. E.g. just decrease it slightly, OR increase it slightly while also lowering damage and/or range in exchange, making them better in "crowd control" but worse at insta-killing everything and wearing down hostile castle walls. 

@Sh0wdown Is it possible to modify mangonel damage against certain types of units/buildings (similar to how arrow projectile damage on units can be changed)? 

---

🔥 **Regarding fire** 🔥 
One thing I have to say is that most vanilla AI castles, as well as many mod castles, do not really take into account fires. One can partition burnable buildings into separated sections, place wells/water pots in sufficient numbers and most importantly, in effective locations, prevent fire spread onto the campfire area, avoid unaccessible burnable buildings (e.g. houses or gallows), etc. 
This way, AI castles with very much increased resistance to fires are possible - and can still be pretty "filled" with buildings, more so than some vanilla castles even, at the same time! 

I really like fire being able to spread and castles to be able to burn down **if the owner doesn't take care**, does not invest time and resources into thinking about how to fireproof the castle and placing wells/waterpots (who require gold and workers). 
Not only is fire a game mechanic that shouldn't be just nerfed into irrelevance, but also are wells and investment that would be pointless too if fire is not an issue anymore. Also, many AI characters using fire ballistae, fire throwers or slaves would lose part of their effectiveness with major nerfs to fire. Lastly, nerfing offensive fire but keeping defensive fire effectiveness the same will make the game more defensive, and quite frankly, defenses do not need a general buff I'd say. It is already difficult enough to light the interior of a well-protected and fortified castle on fire from the outside. 

Fire mechanic changes that I would find okay: 
- Limit fire spreading to a distance of 4 or 5 tiles. Currently, 5 tiles is enough to make fire spreading unlikely, but it still does not prevent it entirely. 
- Currently, fire seems to sometimes spread across buildings that do not burn themselves. Eliminating that would be good imho. 
- When a building has been extinguished, giving it a (slightly longer) immunity to new fires, so that it is a bit easier to end a larger fire, without taking away the means for a fire to get large in the first place. Cause, let's face it, if fire spreading speed was reduced greatly, fires would never ever really have a chance to become large, and would become irrelevant pretty much. Also, hot and dry desert climate does not make fast spreading too unrealistic imho. 
- (maybe) Slightly reduce the "instant spreading" speed of fire through directly adjacent buildings - but do not reduce the (already way slower and most importantly not-instant) spreading speed via sparks, in order to reduce the chances of complete castles being lit on fire at once. 

One thing I see mentioned every now and then are complaints about fire spreading through walls. Well, fire spreads across several tiles by sparks flying off burning buildings and potentially igniting nearby buildings hit by those sparks. 
Sparks do not need to teleport through walls - in most cases, they can simply arc across walls (from a logical point of view). Also, checking for walls in a proper way sounds rather difficult to implement compared to some of the other suggestions / options for changing fires. 

The argument that an entire castle may catch on fire within a second is only partially valid. 
What is true is that fire can spread through several buildings that are directly attached to each other, and sometimes buildings 1 tile apart. 
But 2-tile gaps will prevent this "instant spreading", and any castle owner not willing or able to prevent initial firestarting in his castle can use 2-tile gaps in order to prevent fast fire spreading. 
Yes, you sacrifice a tiny bit of space efficiency, but gain fire resistance in exchange. That, to me, is something that balances different choices for castle design. Also, 5/6-tile gaps (potentially filled with wells/water pots or unburnable buildings/towers, walls, etc.) can prevent spreading in general.

**Comment by PitchNeeded** [comment]
> One more tweak not having been mentioned yet would be to modify the amount of stones per volley, if that is possible. E.g. just decrease it slightly, OR increase it slightly while also lowering damage and/or range in exchange, making them better in "crowd control" but worse at insta-killing everything and wearing down hostile castle walls.

Decreasing the number of stones per volley would be better if this is possible, I am against lowering the range at it would make some of the AI castles very weak against catapults (wolf6 would be easy to beat for example).

**Comment by Sh0wdown** [comment]
In the upcoming v2.13 you will be able to change the fireproof duration of buildings after they were extinguished.
